### PR TITLE
Update dartdoc to v0.18.1 and fix API readme

### LINKED ISSUE #16070: master branch API do not have correct top level description
https://master-docs-flutter-io.firebaseapp.com/

see text beginning:  "This directory contains the documentation generator".

It looks like the magic that was swapping around what is displayed here is broken for some reason.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #16070, #16071.

Note the README.md change in the docs directory in particular.  Most of the rest is cleanup and minor tweaks for adapting to the changes in package handling since 0.17.

Release notes:
https://github.com/dart-lang/dartdoc/releases/tag/v0.18.0
https://github.com/dart-lang/dartdoc/releases/tag/v0.18.1

Demo: [v0.18.1 demo](http://jcollins1.pdx.corp.google.com:8500)

### CODE REVIEW

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -144,6 +143,18 @@ Future<Null> main(List<String> arguments) async {
...
+    for (String arg in args) {
+      if (arg.contains(' ')) {
+        yield "'$arg'";
+      } else {
+        yield arg;
+      }
+    }
+  }
+
+  print('Executing: (cd dev/docs ; $pubExecutable ${quoteArgs(dartdocArgs).join(' ')})');
```
[the remark this answers is not in the document]
Comment (jcollins-g) [reply dev/tools/dartdoc.dart]: Right.  Done.

[the remark this answers is not in the document]
Comment (jcollins-g) [reply dev/tools/dartdoc.dart]: Right, done

File: dev/tools/dartdoc.dart [hunk 2]
```diff
@@ -208,32 +219,15 @@ void createFooter(String footerPath) {
...
+  final List<String> canaries = <String>[
     '$kDocRoot/api/dart-io/File-class.html',
     '$kDocRoot/api/dart-ui/Canvas-class.html',
     '$kDocRoot/api/dart-ui/Canvas/drawRect.html',
-  ];
-  final List<String> canaries = <String>[
     '$kDocRoot/api/flutter_test/WidgetTester/pumpWidget.html',
     '$kDocRoot/api/material/Material-class.html',
     '$kDocRoot/api/material/Tooltip-class.html',
     '$kDocRoot/api/widgets/Widget-class.html',
```
[the remark this answers is not in the document]
Comment (jcollins-g) [reply dev/tools/dartdoc.dart]: connectedTo added.

File: dev/docs/README.md [hunk 3]
```diff
@@ -1,4 +1,8 @@
-This directory contains the documentation generator for
-<https://docs.flutter.io/>. Those docs are updated only for the beta
-branch, see <../bots/docs.sh>. You can find docs for the master branch
-staged at <https://master-docs-flutter-io.firebaseapp.com/>.
+A new mobile app SDK to help developers and designers build modern mobile apps for iOS and Android. Flutter is an open-source project currently in beta.
```
[the remark this answers is not in the document]
Comment (tvolkert) [reply dev/docs/README.md]: Ah, makes sense.  In that case, I'd crib the verbiage off the main flutter.io site:

> Flutter is Google’s mobile UI framework for crafting high-quality native interfaces on iOS and Android in record time. Flutter works with existing code, is used by developers and organizations around the world, and is free and open source.

[the remark this answers is not in the document]
Comment (jcollins-g) [reply dev/docs/README.md]: I had borrowed from the top level readme in the flutter repo, but I'm OK with it either way.  Done.

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**